### PR TITLE
feat: Screen Share improvement for Group Calls [WPB-10134]

### PR DESCRIPTION
**Pull request by EnricoSchw**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10134" title="WPB-10134" target="_blank">StoryWPB-10134</a>  [AVS] Change Screen Share constrains for better Screen Share quality in Group Call
  </td></table>
  <br />
 

###  This is an improvement to the readability of screen sharing:

- I have removed the downscaling on the sender side.
- I mark screen share videos as "All Content Detailed."
- If someone receives a 4K screen share on their 1080p monitor, it will naturally be downscaled on the receiver's side. For this case, I have added a zoom feature.
- We now send screen shares at 5 fps by default.

## Screenshots/Screencast (for UI changes)

Bildschirmfoto 2024-12-06 um 17 48 37

@WolfiWire 

I think you'll have an opinion on this, but I'd like to adjust the zoom feature in a fallow up PR. I hope this is ok 
(These PRs always take so long :) )

## Checklist

- [x ] mentions the JIRA issue in the PR name (Ex. [WPB-XXXX])
- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/calling/GroupVideoGridTile.tsx [hunk 1]
```diff
@@ -105,8 +108,29 @@ const GroupVideoGridTile: React.FC<GroupVideoGridTileProps> = ({
...
   };
 
+  const handleZoomClick = () => {
+    if (isZoomedIn) {
+      setIsZoomedIn(false);
+    } else {
+      setIsZoomedIn(true);
+    }
+  };
+
```
Comment (thisisamir98) [inline src/script/components/calling/GroupVideoGridTile.tsx]:
```suggestion
  const handleZoomClick = () => {
    setIsZoomedIn(prev => !prev); 
```

File: src/script/components/calling/GroupVideoGridTile.tsx [hunk 2]
```diff
@@ -105,8 +108,29 @@ const GroupVideoGridTile: React.FC<GroupVideoGridTileProps> = ({
...
   const participantNameColor = getParticipantNameColor({isActivelySpeaking, isAudioEstablished});
 
+  const actionItem = !minimized && sharesScreen && (
+    <button
+      data-uie-name="item-grid"
+      data-user-id={participant?.user.id}
+      className="group-video-grid__element__action_icon"
+      onClick={handleZoomClick}
+      onKeyDown={handleZoomClick}
```
Comment (thisisamir98) [inline src/script/components/calling/GroupVideoGridTile.tsx]: two are unnecessarily, only onClick should be enough

File: src/script/team/TeamState.ts [hunk 3]
```diff
@@ -117,9 +117,7 @@ export class TeamState {
       () => this.teamFeatures()?.mls?.config.protocolToggleUsers.includes(this.userState.self().id) ?? false,
     );
 
-    this.isConferenceCallingEnabled = ko.pureComputed(
-      () => this.teamFeatures()?.conferenceCalling?.status === FeatureStatus.ENABLED,
-    );
+    this.isConferenceCallingEnabled = ko.pureComputed(() => true);
```
Comment (thisisamir98) [inline src/script/team/TeamState.ts:120]: if it's gonna be always true I suggest a bigger refactor to just get rid of this property every where, but for now we could at least set it to an always true observable maybe

```suggestion
    this.isConferenceCallingEnabled = ko.observable(true);
```

Comment (EnricoSchw) [reply src/script/team/TeamState.ts:120]: O good finding.. ❤️  Thx. this is a hack to activate group call.. This should never go to staging. I will remove this!

File: src/script/components/calling/GroupVideoGridTile.tsx (hunk 2 above)
Comment (EnricoSchw) [reply src/script/components/calling/GroupVideoGridTile.tsx]: You use both events in many places... I thought it was a workaround.  I will remove it

**Review by thisisamir98 — APPROVED** [review]

**Pull request merged**